src/execution_plan/optimizations/compact_filters.c

```diff
@@ -0,0 +1,52 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "compact_filters.h"
+#include "../ops/op_filter.h"
+#include "../../filter_tree/filter_tree.h"
+
+// Try to compact a filter.
+static inline bool _compactFilter(OpBase *op) {
+	assert(op->type == OPType_FILTER);
+	OpFilter *filter_op = (OpFilter *)op;
+	return FilterTree_Compact(filter_op->filterTree);
+}
+
+// In case the compacted filter resolved to true, remove it from the execution plan.
+static void _removeTrueFilter(ExecutionPlan *plan, OpBase *op) {
+	assert(op->type == OPType_FILTER);
+	OpFilter *filter_op = (OpFilter *)op;
+	FT_FilterNode *root = filter_op->filterTree;
+	assert(root->t == FT_N_EXP);
+	SIValue bool_val = AR_EXP_Evaluate(root->exp.exp, NULL);
```

Comment:
Can we evaluate expressions containing variables? e.g. `n.v+2`

Reply:
done